src/UnitTests/Core/Impl/MarkupUtils.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Python.Core.Text;
+
+namespace Microsoft.Python.UnitTests.Core {
```

Comment:
In which test assemblies is this going to be used? Does it make more sense for it to end up in `Microsoft.Python.LanguageServer.Tests` over the core unit test? If this is generically useful enough to not matter, then no big deal.

Reply:
I think it is generally useful since you can annotate test code itself to mark stuff like where error should be associated or marking a variable to bind and see types and etc.